Calculator/CalculatorViewController.swift

```diff
@@ -12,31 +12,47 @@ class CalculatorViewController: UIViewController {
     
     @IBOutlet weak var outputLabel: UILabel!
     
+    var brain: CalculatorBrain?
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        brain = CalculatorBrain()
     }
     
     // MARK: - Action Handlers
     
     @IBAction func operandTapped(_ sender: UIButton) {
-        
+        if let buttonTitle = sender.currentTitle {
+            outputLabel.text = brain?.addOperandDigit(buttonTitle)
+        }
     }
     
     @IBAction func operatorTapped(_ sender: UIButton) {
-        
+        if let buttonTitle = sender.currentTitle {
+            brain?.setOperator(buttonTitle)
+        }
+    }
+    
+    @IBAction func unaryOperatorTapped(_ sender: UIButton) {
+        if let buttonTitle = sender.currentTitle, let result = brain?.calculateUnaryOperation(buttonTitle) {
+            outputLabel.text = result
+        }
     }
     
     @IBAction func equalTapped(_ sender: UIButton) {
-        
+        if let result = brain?.calculateIfPossible() {
+            outputLabel.text = result
+        }
     }
     
     @IBAction func clearTapped(_ sender: UIButton) {
-        
+        clearTransaction()
+        outputLabel.text = "0"
     }
     
     // MARK: - Private
     
     private func clearTransaction() {
-        
```

Comment:
Here you would want to set the brain to nil. To make sure that old instances of the CalculatorBrain arent being stored in memory. 

`brain = nil`